cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +34,67 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_main_package_name_and_version(package_dir, dependencies)
     return []
+
+
+def _get_main_package_name_and_version(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """
+    Get main package name and version.
+
+    The main package is the package that is being processed by cachi2.
+    Not any of its dependencies.
+    """
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    name_and_version = _get_name_and_version_from_path_deps(path_dependencies)
+    if name_and_version is not None:
+        return name_and_version
+
+    log.info("Failed to infer name and version from gemspec file")
+
+    # fallback to origin remote
+    try:
+        name = _get_repo_name_from_origin_remote(package_dir)
+    # if the git repository does not have an origin remote
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Failed to infer package name from origin remote",
+            solution=(
+                "Please specify package name and version in a way that Cachi2 understands,\n"
+                "or make sure that the directory Cachi2 is processing is a git repository with\n"
+                "an 'origin' remote, in which case Cachi2 will infer the package name from the remote URL."
+            ),
+        )
+
+    return name, None
+
+
+def _get_name_and_version_from_path_deps(
+    path_dependencies: list[PathDependency],
+) -> Optional[tuple[str, str]]:
+    """
+    Extract name and version from path dependencies in the Gemfile.lock.
+
+    If the gemspec file exists in the package directory and is explicitly defined in the Gemfile,
+    then the Gemfile.lock will contain a path dependency representing the gem in the package directory.
+    """
+    for dep in path_dependencies:
+        if dep.path == ".":
+            return dep.name, dep.version
+
+    return None
+
+
+def _get_repo_name_from_origin_remote(package_dir: RootedPath) -> str:
+    """Extract repository name from git origin remote in the package directory."""
+    repo_path = get_repo_id(package_dir.root).parsed_origin_url.path.removesuffix(".git")
+    repo_name = Path(repo_path).name
+    package_subpath = package_dir.subpath_from_root
+
+    resolved_path = Path(repo_name).joinpath(package_subpath)
+    normalized_path = canonicalize_name(str(resolved_path).replace("/", "-"))
```

Comment:
Using `canonicalize_name` doesn't seem correct to me in this case. We're applying [PEP 503](https://peps.python.org/pep-0503/#normalized-names) onto something that not only doesn't conform to it, but also doesn't have a formally defined set of acceptable characters. I could only find https://guides.rubygems.org/name-your-gem/ which allows usage of both `-` and `_` and we should respect it in the SBOM as well to avoid any confusion on the user side.